proto/message_contents/public_key.proto

```diff
@@ -0,0 +1,54 @@
+// Protos for XMTP Messages and their headers
+syntax = "proto3";
+
+package xmtp.message_contents;
+
+option go_package = "github.com/xmtp/proto/go/message_contents";
+
+// Signature represents a generalized public key signature,
+// defined as a union to support cryptographic algorithm agility.
+message Signature {
+    // The format for an ECDSA signature bytes and the recovery bit
+    message ECDSACompact {
+        bytes bytes = 1; // compact representation [ R || S ], 64 bytes
+        uint32 recovery = 2; // recovery bit
+    }
+    oneof union {
+        ECDSACompact ecdsa_compact = 1;
+    }
+}
+
+// PublicKey represents a generalized public key,
+// defined as a union to support cryptographic algorithm agility.
+message PublicKey {
+    // The key bytes
+    message Secp256k1Uncompressed {
+        // uncompressed point with prefix (0x04) [ P || X || Y ], 65 bytes
+        bytes bytes = 1; 
+    }
+    uint64 timestamp = 1;
+    optional Signature signature = 2;
+    oneof union {
+        Secp256k1Uncompressed secp256k1_uncompressed = 3;
+    }
+}
+
+// PublicKeyBundle packages the cryptographic keys associated with a wallet,
+// both senders and recipients are identified by their key bundles.
+message PublicKeyBundle {
+    PublicKey identity_key = 1;
+    PublicKey pre_key = 2;
+}
+
+// The PublicKeyBundle of the user on the network.
+// Used to derive shared secrets for encrypted messaging
+message ContactBundleV1 {
```

Comment:
Do we actually want to break ContactBundle into it's own file too while we are at it?

Reply:
I'm game, do you mean just the ContactBundle or the PublicKeyBundle as well? If the latter then we should probably break out PrivateKeyBundle as well for consistency. I think I'd slightly prefer to keep it to just the ContactBundle.